tests/test.rs

```diff
@@ -2039,3 +2039,52 @@ fn test_integer128() {
         ),
     ]);
 }
+
+#[test]
+fn test_raw_value() {
+    #[derive(Serialize, Deserialize)]
+    struct Wrapper<'a> {
+        a: i8,
+        #[serde(borrow)]
+        b: RawValue<'a>,
+        c: i8,
+    };
+
+    let wrapper_from_str =
+        serde_json::from_str::<Wrapper>(r#"{"a": 1, "b": {"foo": 2}, "c": 3}"#).unwrap();
+    assert_eq!(r#"{"foo": 2}"#, wrapper_from_str.b.as_ref());
+
+    let wrapper_from_reader = serde_json::from_reader::<_, Wrapper<'static>>(
```

Comment:
This is a compiler bug, this line shouldn't compile. I filed https://github.com/rust-lang/rust/issues/54302 to follow up. Still thinking about what that means for how we want to expose RawValue.

Reply:
Hah, of course, not sure what I thought this would do!

Does that mean we want to have an owned and a borrowed variant of `RawValue`?

Reply:
I believe so. I opened #485 to experiment with that design. Let me know what you think.